File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size()-1;
        int MAX = 0;
        int area = 0;
        int distrance;
        int level;
        while(left<right){
            distrance = right-left;
            level = min(height[right], height[left]);
            area = distrance*level;
            MAX = max(MAX, area);
            if (height[right]<height[left]){
                right--;
            }else{
                left++;
            }
        
        }
        return MAX;
        
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int right = n-1;
        int left  = 0;
        int area = 0;
        int ans = 0;
        while(left<right){
            area = (right - left) * Math.min(height[left], height[right]);
            ans = Math.max(ans, area);
            if(height[left]<height[right]){
                left++;
            }else{
                right--;
            }
        }
        return ans;
    }
}